### ISSUE TITLE
EU features list

### ISSUE THREAD

**Issue by NickSto** [galaxyproject/galaxy-hub#1117]
Here's where we can attempt to list all the individual features present on galaxyproject.eu which we may want to re-implement in the unified Hub.

**Comment by NickSto** [comment]
## Feature
### It provides the middle pane for many Galaxy instances
Each of the `index-*.md` files in the root directory generates a page that's hotlinked from an iframe in a Galaxy instance.

For instance:  
`index-rna.md` generates
https://usegalaxy-eu.github.io/index-rna.html, which is the center pane of
https://rna.usegalaxy.eu/

**Comment by NickSto** [comment]
## Feature
### Generate pages without the window dressing (NavBar)
There are many pages on the .eu site which are just content - no navbar or other chrome.
It's necessary to, for instance, provide a middle pane for a Galaxy instance.
They also seem to generate other content-only pages for inclusion in Galaxy instances: https://github.com/usegalaxy-eu/website/#duplication

**Comment by NickSto** [comment]
## Feature
### Automatically inferring date from filename
There are some files with no date in their metadata which still end up with a date presented on the page.
For example: [_posts/2021-02-14-smorgasbord-starts.md](https://raw.githubusercontent.com/usegalaxy-eu/website/master/_posts/2021-02-14-smorgasbord-starts.md) yields https://galaxyproject.eu/posts/2021/02/14/smorgasbord-starts/
I can only assume it's inferred from the filename?

**Comment by NickSto** [comment]
## Feature
### Tweet Wall?
https://github.com/usegalaxy-eu/website/tree/master/tweet-wall

**Comment by NickSto** [comment]
## Feature
### Widgets?
https://github.com/usegalaxy-eu/website/tree/master/widgets
https://github.com/usegalaxy-eu/website/blob/master/widgets/index.md

**Comment by NickSto** [comment]
## Feature
### jekyll-feed plugin
The `_config.yml` shows they have [jekyll-feed](https://github.com/jekyll/jekyll-feed), and indeed there's a [/feed.xml](https://galaxyproject.eu/feed.xml) live on the site.

**Comment by NickSto** [comment]
## Feature
### tools.yml
Not sure what it's doing
https://github.com/usegalaxy-eu/website/blob/master/_data/tools.yml

**Comment by NickSto** [comment]
## Feature
### Events ical format
The `_events_ical` directory is there to render the list of events in `.ics` format.
It's actually a link to `_events`, meaning all events will be duplicated in ical format.
- The template is `_layouts/event-ical.ics`.
Still not sure what url this ends up at, though.

**Comment by bgruening** [comment]
> ## Feature
> ### Automatically inferring date from filename
> 
> There are some files with no date in their metadata which still end up with a date presented on the page. For example: [_posts/2021-02-14-smorgasbord-starts.md](https://raw.githubusercontent.com/usegalaxy-eu/website/master/_posts/2021-02-14-smorgasbord-starts.md) yields https://galaxyproject.eu/posts/2021/02/14/smorgasbord-starts/ I can only assume it's inferred from the filename?

Posts are usually retrieving the data from the filename. Events on the other side do take the dates from metadata. However, both take the date from the filename to hide posts and events which filename is in the future. This enables us to create events and posts that are only published automatically in the future.
But Nick this does not need to stay like that. If you think the data should be metadata also for posts, we can do that, We can script that, change all our markdown when we migrate. It would just be good if we can preserve the links, or redirect them. A few posts are hightly read.



> ## Feature
> ### Tweet Wall?
> 
> https://github.com/usegalaxy-eu/website/tree/master/tweet-wall

This we used during GCC and displayed that on a big screen. I think its a cool feature, but lets move this to a separate issue, its not needed for the migration.

> ## Feature
> ### Widgets?
> 
> https://github.com/usegalaxy-eu/website/tree/master/widgets https://github.com/usegalaxy-eu/website/blob/master/widgets/index.md

Those widgets we use Galaxy webhooks. So if you submit a Galaxy tool, you will see once in a while the latest events/news... Just a different, more compact view of events/news.

> ## Feature
> ### jekyll-feed plugin
> 
> The `_config.yml` shows they have [jekyll-feed](https://github.com/jekyll/jekyll-feed), and indeed there's a [/feed.xml](https://galaxyproject.eu/feed.xml) live on the site.

Very important for a few users!

> ## Feature
> ### Events ical format
> 
> The `_events_ical` directory is there to render the list of events in `.ics` format. It's actually a link to `_events`, meaning all events will be duplicated in ical format.
> 
>     * The template is `_layouts/event-ical.ics`.
> 
> 
> Still not sure what url this ends up at, though.

webcal://galaxyproject.eu/event/2021-12-23-GalaxyWS_SFB/event.ics.html

You see that on each event on the top:

![image](https://user-images.githubusercontent.com/469983/150632335-3f8a861c-bd4f-4c83-92c2-8673b2d85a79.png)

Thanks for working on this! There are several sites, like AU, BE, Africa that are using this website, so all time you spent here will enable the others sites to migrate as well. Its worth it I think! Thanks!

**Comment by bgruening** [comment]
> ## Feature
> ### [tools.yml](https://github.com/usegalaxy-eu/website/blob/master/_data/tools.yml)
> 
> Looks like a database mapping tool names/ids to toolshed locations. Seems to only be used to generate urls to tools in index-*.md middle panes (see [index-rna.md](https://usegalaxy-eu.github.io/index-rna.html#available-tools))
> 
>     * .md page uses an `include` like `{% include tool.html id="CoFold" %}`
> 
>     * Then `_includes/tool.html` queries the tools.yml database with `{%- for tool in site.data.tools -%}`
> 
>     * Then it generates a link to the tool with a template like `https://usegalaxy.eu/root?tool_id={{ tool[1] }}`
> 
> 
> Strangely, there's a [/tools](https://galaxyproject.eu/tools) page, but that's hardcoded in HTML in `tools.md`, and doesn't seem to be generated by `tools.yml`. Perhaps because it's grouped into categories, which don't exist in `tools.yml`.

Upps, missed that one.
The tools.yml is just a convenience mapping of keys to proper tool-urls. This can be used for example on subdomains to link conveniently tools, e.g. in those markdown tables, without copying the URL every time AND more important, dependent on the subdomain the link is __`https://singlecell.usegalaxy.eu/root?tool_id={{ tool[1] }}`__ or __`https://rna.usegalaxy.eu/root?tool_id={{ tool[1] }}`__

This page https://galaxyproject.eu/tools is completely different and it is autogenerated every week, it's a plain HTML.
See here: https://github.com/usegalaxy-eu/website/blob/master/.github/workflows/update-tools.yml You can simply take this GitHub action over and run this for all Galaxy instances.

**Comment by NickSto** [comment]
# Feature
## Expanded Font Awesome icon set

We seem to not have as much Font Awesome support as .eu. Some icons work out of the box (`fa-laptop`, `fa-list-ul`), but others show up as an unrecognized unicode character (`fa-slideshare`) and still others don't show up at all (`fa-video-camera`).

You can see examples at `/events/2021-10-sars-cov-2-training/`.

**Comment by NickSto** [comment]
## Feature
### Display `supporters` in footer

.eu has a `supporters` metadata key which seems to add a "Supported by" footer with organization logos beneath.

See: https://galaxyproject.eu/posts/2022/04/24/tool-update/

**Comment by NickSto** [comment]
## Feature
### Structured article metadata
As described in #1315, certain metadata fields in .eu posts contain structured data instead of plain strings. We need to parse them and display them properly:
#### `organiser`
These are often dicts which contain keys like `name` and `email`.
#### `location`
These are often dicts which contain keys like `city`, `country`, and `geo` (which itself contains a dict of GPS coordinates).
#### `author`
28 posts in `_posts` (and none in `_events`) contain this key. It'd be good to display this in the headers of news posts.

**Comment by NickSto** [comment]
## Feature
### Automatic article teases
.eu seems to automatically generate `tease`s for their posts from the first bit of the content. There's an automatic function to do that in Gridsome.

When implementing it, though, it'd be good to somehow avoid displaying auto-generated teases in the article header, like we currently do for our manual teases. Because then it looks repetitive and weird.

**Comment by bgruening** [comment]
> ## Feature
> ### Automatic article teases
> 
> .eu seems to automatically generate `tease`s for their posts from the first bit of the content. There's an automatic function to do that in Gridsome.
> 
> When implementing it, though, it'd be good to somehow avoid displaying auto-generated teases in the article header, like we currently do for our manual teases. Because then it looks repetitive and weird.

It's debatable if this feature is good or bad. I could argument in both sides I guess. If you think we should drop it we need to bite the bullet and add tease to our articles.

**Comment by NickSto** [comment]
> It's debatable if this feature is good or bad. I could argument in both sides I guess. If you think we should drop it we need to bite the bullet and add tease to our articles.

From my side, I'd certainly prefer manual teases. But it's just a little extra work to auto-generate them.

**Comment by NickSto** [comment]
## Feature
### RSS feed

Reproduce https://galaxyproject.eu/feed.xml

Hopefully we can just use a Gridsome plugin like https://gridsome.org/plugins/gridsome-plugin-feed

One potential hitch is that if we want to give people a drop-in replacement for the .eu feed, we'd want to filter for only `eu` posts.

**Comment by NickSto** [comment]
## Feature
### Hiding footer

See `_events/2020-10-22-RNA-Seq-Iran2020.md`:
```
# special hiding of footer since we want to do it manually.
hidefooter: true
```

**Comment by NickSto** [comment]
## Usage
How many of these features are actually used or relied on?
Here I'm collecting specifics on what data would help deciding whether each feature warrants actually implementing.

### Widgets
What are the traffic stats for:
https://galaxyproject.eu/widgets/news.html
https://galaxyproject.eu/widgets/events.html

### Events ical format
What are the traffic stats for:
https://galaxyproject.eu/events/*/event.ics.html
https://galaxyproject.eu/event/*/event.ics.html

### Pages without window dressing
What are the traffic stats for:
https://galaxyproject.eu/events/*/plain.html
https://galaxyproject.eu/event/*/plain.html
https://galaxyproject.eu/posts/*/*/*/plain.html

**Comment by bgruening** [comment]
> What are the traffic stats for:
> https://galaxyproject.eu/widgets/news.html
> https://galaxyproject.eu/widgets/events.html

Those are very high, we use them inside Galaxy webhooks. This means those are displayed when people submit Galaxy jobs.

**Comment by bgruening** [comment]
> What are the traffic stats for:
https://galaxyproject.eu/events/*/event.ics.html
https://galaxyproject.eu/event/*/event.ics.html

No idea, not sure if we know. Is this hard to generate? We should actually create a public GCalender out of that, or add this automatically to an existing one.

**Comment by beatrizserrano** [comment]
> > What are the traffic stats for:
> > https://galaxyproject.eu/events/*/event.ics.html
> > https://galaxyproject.eu/event/*/event.ics.html
> 
> No idea, not sure if we know. Is this hard to generate? We should actually create a public GCalender out of that, or add this automatically to an existing one.

@dannon is this what you worked on? I think that the only missing part was to add them to the existing `GalaxyEvents` calendar

**Comment by NickSto** [comment]
> No idea, not sure if we know. Is this hard to generate? We should actually create a public GCalender out of that, or add this automatically to an existing one.

Yeah, so Dannon was working on an ics feed which would contain all events. It's a little different, but maybe that's close enough to say it supplants the individual event ics files?

**Comment by NickSto** [comment]
> Those are very high, we use them inside Galaxy webhooks. This means those are displayed when people submit Galaxy jobs.

@bgruening Very good to know. Are you sure these are the exact urls that are used, though? I just tried it myself (on usegalaxy.eu) and it showed https://usegalaxy-eu.github.io/galaxy/news.html, not https://galaxyproject.eu/widgets/news.html

**Comment by NickSto** [comment]
## Feature

### Inter-post links

Each news (but not event) post has links to the previous and next posts so users can navigate between them without going back to the list of all events.

**Comment by dannon** [comment]
Yep, the .ics feed is live at https://galaxyproject.org/events/calendar.ics and will include any 'event' type content item; I think what @beatrizserrano is mentioning was piping all of those events into the existing `GalaxyEvents` public google calendar.  That turned out to be nontrivial and potentially messy (like for changes to events) to keep in sync.  Still looking for better ways to solve that one.

**Comment by NickSto** [comment]
Re: the "add to calendar" feature, it looks like we can also give an "add to Google Calendar" button without any ics file. Instead all we need is to compose a link to a particular Google Calendar url.

For example: `http://www.google.com/calendar/event?action=TEMPLATE&details=We+will+have+a+presentation+on+Galaxy+on+AnVIL+by+Mike+Schatz&text=Galaxy+Community+Call%3A+Galaxy+on+AnVIL&dates=20220526T150000Z%2F20220526T160000Z`

This breaks down to `http://www.google.com/calendar/event` with these query parameters:
| parameter | value |
|---------------|---------|
| action |  TEMPLATE |
| details | We+will+have+a+presentation+on+Galaxy+on+AnVIL+by+Mike+Schatz |
| text |    Galaxy+Community+Call:+Galaxy+on+AnVIL |
| dates |   20220526T150000Z/20220526T160000Z |

**Comment by NickSto** [comment]
Bea gave some traffic data that shed some light on some of the questions above:

> ### Widgets
> 
> What are the traffic stats for:
> https://galaxyproject.eu/widgets/news.html
> https://galaxyproject.eu/widgets/events.html

/widgets/news.html is the 8th most visited page in the last month. I didn't see events.html in the top 45, though.

> ### Pages without window dressing
> 
> What are the traffic stats for:
> https://usegalaxy-eu.github.io/galaxy/news.html
> https://usegalaxy-eu.github.io/galaxy/events.html

/galaxy/events.html and /galaxy/news.html are the 2nd and 3rd most visited pages, far above the versions with navbars.

Here are all the versions of the events and news pages in the top 45 most visited pages:

| rank | page |
|------|------|
| 2    | /galaxy/events.html |
| 3    | /galaxy/news.html |
| 8    | /widgets/news.html |
| 27   | /news |
| 39   | /news.html |
| 41   | /events |

**Comment by NickSto** [comment]
Feature
----------

### Filtering posts by tag

It looks like you can filter posts by tag with query parameters like so: https://galaxyproject.eu/news?tag=UseCase

**Comment by NickSto** [comment]
Feature
----------

### The `doi` metadata field

It looks like when a post includes a `doi` field, they generate a link to the article next to the tags in the header. See: https://galaxyproject.eu/posts/2022/05/27/paper-tnbc/
